projects/distributed-tracing/src/shared/dashboard/widgets/table/table-widget-renderer.component.ts

```diff
@@ -226,6 +298,27 @@ export class TableWidgetRendererComponent
     this.checkboxFilterSubject.next(tableFilter ? [tableFilter] : []);
   }
 
+  public onSelectChange(changed: SelectChange): void {
+    const selectFilterModel = this.model
+      .getSelectFilterOptions()
+      .find(option => option.attribute === changed.value.key);
+
+    if (selectFilterModel === undefined) {
+      return; // This shouldn't happen
+    }
+
+    const existingSelectFiltersWithChangedRemoved = this.selectFilterSubject
+      .getValue()
+      .filter(f => f.field !== changed.value.key);
+    const changedSelectFilter = selectFilterModel.getTableFilter(changed.value.value);
+
+    const selectFilters = [...existingSelectFiltersWithChangedRemoved, changedSelectFilter].filter(
+      f => f.value !== undefined
```

Comment:
does the undefined filtering mean we can't switch back to unfiltered (isn't undefined the "All Services" option) ?

Reply:
Correct. The select option returning a value of `undefined` signals to the widget to drop that filter, so we remove it from the request all together.

Reply:
I'm confused - I would expect to not use that filter in the request, but it should still remain in the dropdown list if I want to return to see all apis across services. The behavior I read/you describe sounds like it is removed from the selection list once an option is chosen, never to return.